runtime/src/main/java/org/corfudb/runtime/object/transactions/TransactionBuilder.java

```diff
@@ -27,7 +28,7 @@
     /** For snapshot transactions, the address the
      * snapshot will start at.
      */
-    public long snapshot = -1L;
+    public long snapshot = Address.NON_ADDRESS;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make snapshot a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)